src/apps/answers/service.py

```diff
@@ -276,6 +280,37 @@ async def _create_answer(self, applet_answer: AppletAnswerCreate) -> AnswerSchem
         )
         return answer
 
+    async def validate_take_now(
+        self,
+        applet_id: uuid.UUID,
+        target_subject_id: uuid.UUID | None = None,
+        source_subject_id: uuid.UUID | None = None,
+        activity_or_flow_id: uuid.UUID | None = None,
+    ):
+        assert self.user_id
+        subject_crud = SubjectsCrud(self.session)
+
+        respondent_subject = await subject_crud.get_user_subject(user_id=self.user_id, applet_id=applet_id)
+        if not respondent_subject or not respondent_subject.soft_exists():
+            raise TakeNowNoAccessApplet()
+
+        if target_subject_id:
+            target_subject = await subject_crud.get_by_id(target_subject_id)
+            if not target_subject or not target_subject.soft_exists() or target_subject.applet_id != applet_id:
+                raise TakeNowInvalidTargetSubject()
+
+        if source_subject_id:
+            source_subject = await subject_crud.get_by_id(source_subject_id)
+            if not source_subject or not source_subject.soft_exists() or source_subject.applet_id != applet_id:
+                raise TakeNowInvalidSourceSubject()
+
+        if activity_or_flow_id:
+            activity_or_flow = await ActivitiesCRUD(self.session).get_by_applet_id_and_activity_id(
+                applet_id=applet_id, activity_id=activity_or_flow_id
+            )
+            if not activity_or_flow:
+                raise TakeNowInvalidActivityOrFlow()
```

Comment:
I don't think this validation will work for activity flows. I think you could update it to use `FlowsCRUD.get_by_applet_id_and_flow_id` when the activity validation fails, since we don't know whether the ID is an activity ID or an activity flow ID

Reply:
done